### PR TITLE
Chapter 2: Workflow plan

### PR DESCRIPTION
**Pull request by caalo**
#### What changes are being implemented in this Pull Request?

Draft of workflow plan created!

### Tell potential reviewers what kind of feedback you are soliciting.

Need Sita's workflow diagram and edits on what the test samples are exactly. 

### New Content Checklist

- [ ] New content/chapter is in an Rmd file with [this kind of format and headers](https://github.com/jhudsl/OTTR_Template/blob/main/02-chapter_of_course.Rmd).

- [ ] New content/chapter contains learning objectives.

- [ ] [Bookdown successfully re-renders and any new content files have been added to the _bookdown.yml](https://github.com/jhudsl/OTTR_Template/wiki/Publishing-with-Bookdown).

- [ ] [Spell check runs successfully](https://www.ottrproject.org/customize-robots.html#Spell_checking)).

- [ ] Any newly necessary packages that are needed have been added to the [Dockerfile and image](https://www.ottrproject.org/customize-docker.html).

- [ ] Images are in the [correct format for rendering](https://www.ottrproject.org/writing_content.html#set-up-images).

- [ ] Every new image has [alt text and is in a Google Slide](https://www.ottrproject.org/writing_content.html#Accessibility).

- [ ] Each slide is described in the notes of the slide so learners relying on a screen reader can access the content. See https://lastcallmedia.com/blog/accessible-comics for more guidance on this.

- [ ] The color palette choices of the slide are contrasted in a way that is friendly to those with color vision deficiencies.
You can check this using [Color Oracle](https://colororacle.org/).

### CODE REVIEW

File: index.Rmd [hunk 1]
```diff
@@ -19,7 +19,7 @@ This WDL workflow development guide shows a bioinformatics workflow developer ho
 
 ## Target Audience
 
-The course is intended for first time developers of the [WDL workflow language](https://github.com/openwdl/wdl), who wants to iteratively develop a WDL bioinformatics workflow. The audience should be able to run a WDL workflow on a computing engine of their choice, such as Cromwell, miniWDL, or a cloud computing environment such as Terra, AnVIL, or Dockstore. 
+The course is intended for first time developers of the [WDL workflow language](https://github.com/openwdl/wdl), who wants to iteratively develop a WDL bioinformatics workflow. The audience should be able to comprehend WDL syntax, and should be able to run a WDL workflow on a computing engine of their choice, such as Cromwell, miniWDL, or a cloud computing environment such as Terra, AnVIL, or Dockstore.
```
Comment (aofarrel) [inline index.Rmd:22]: >  and should be able to run a WDL workflow on a computing engine of their choice, such as Cromwell, miniWDL, or a cloud computing environment such as Terra, AnVIL, or Dockstore. 

I think this is fair for what we're trying to do, but it does contradict what was said in the intro section in the previous PR. Should I reword that part of the introduction?

File: 02-workflow-plan.Rmd [hunk 2]
```diff
@@ -2,9 +2,34 @@
...
 
-## The workflow
+## Somatic mutation calling workflow
 
-## The test data
\ No newline at end of file
+The workflow that we will build is a somatic mutation calling workflow on paired cancer-normal samples. Our workflow is inspired from a somatic mutation calling [workflow](https://github.com/FredHutch/tg-wdl-LILAC-workflow/blob/master/GATK4.1.4.0/GATK4.1.4.0_manta.wdl) by the [Berger Lab](https://research.fredhutch.org/berger/en.html) at the Fred Hutch Cancer Center, and a somatic mutation calling [workflow](https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-) by the GATK team at the Broad Institute. The workflow takes in pairs of cancer and normal samples from sequencing in the FASTQ format and aligns the sequencing reads of each sample to the hg19 human reference genome, marks PCR duplicates, and recalibrates base quality. Which a pair of cancer and normal samples have reached this step, the workflow performs somatic mutation calling on them.
```
Comment (aofarrel) [inline 02-workflow-plan.Rmd:11]: > Which a pair

Possible typo?

File: 02-workflow-plan.Rmd [hunk 3]
```diff
@@ -2,9 +2,34 @@
...
+
+2.  End-to-end testing: When we connect all the tasks together to form a workflow, we test that the workflow running end-to-end is correct.
+
+Here are some guidelines for any form of testing:
+
+-   The data you use for testing is representative of "real" data.
+
+-   You have an expectation of what the resulting output is *before* you run your workflow on it. It can be as specific as a MD5 checksum, or vague such as a certain file format.
+
+-   The process is quick to run, at the minutes level. This often means using a small subset of actual data.
```
Comment (aofarrel) [inline 02-workflow-plan.Rmd:31]: I would change the first sentence to "The process is quick to run, ideally in the range of just a few minutes." 95% of the time you should aim for a quick run, but there are some rare cases where what you are testing truly cannot be properly tested end-to-end properly in a few minutes. I consider "tests run within a few minutes" to be very useful, but I don't want users to think that's as important as the data being representative.

(Example: One workflow I worked on took in [dbGaP](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/about.cgi) data using a special "drs" protocol. The testing needed to be able to check that the drs protocol was working, and for a while there were no downsampled samples I could test the drs integration with.)

Comment (caalo) [reply 02-workflow-plan.Rmd]: Sounds good!

File: index.Rmd (hunk 1 above)
Comment (aofarrel) [reply index.Rmd:22]: Sorry, I should have clarified -- in part 1 of the tutorial, we mention that being able to run a WDL isn't a requirement for this course. The first time I read this passage here in part 2, I thought "and should be able to run a WDL workflow on a computing engine" contradicted that. But reading it again, it really isn't a contradiction.

**Comment by aofarrel** [comment]
I think we can clarify how the pipeline works by rewording this:

> The workflow takes in pairs of cancer and normal samples from sequencing in the FASTQ format and aligns the sequencing reads of each sample to the human reference genome, marks PCR duplicates, and recalibrates base quality. When a pair of cancer and normal samples have reached this step, the workflow performs somatic mutation calling on them.

Into something like this:

> The workflow takes in FASTQ-formatted sequencing data from pairs of tumor and normal samples. The workflow then aligns the FASTQs of each sample to the human reference genome, marks PCR duplicates, and recalibrates base quality. As a final step, the workflow performs somatic mutation calling on the tumor and normal samples, comparing the two of them to see how the tumor and normal samples differ from each other.

If that's overkill, maybe something like this:

> The workflow takes in FASTQ-formatted pairs of tumor and normal samples. These FASTQs are then aligned to the human reference, followed by marking of PCR duplicates, and base quality recalibration. As a final step, the workflow performs somatic mutation calling on the tumor and normal samples.

**Comment by caalo** [comment]
@aofarrel yeah that's more clear, thanks! Just made the change.

**Comment by caalo** [comment]
@sitapriyamoorthi when you have time can you add the details about the test samples and workflow diagram?

File: 02-workflow-plan.Rmd [hunk 4]
```diff
@@ -2,9 +2,34 @@
 ottrpal::set_knitr_image_path()
 ```
 
-# Defining a workflow plan: Introduce somatic mutation calling workflow
+# Defining a workflow plan
 
+Our WDL guide will center around building a workflow from scratch. As we build out this workflow step-by-step, you will see what strategies and resources are used to develop a workflow that is iterative, reproducible, and efficient in terms of time and resource used. The goal is to use this workflow to illustrate common lessons in writing WDL workflows.
 
-## The workflow
+## Somatic mutation calling workflow
```
Comment (sitapriyamoorthi) [inline 02-workflow-plan.Rmd:9]: The example workflow:  Somatic mutation calling  using MuTect2

**Review by sitapriyamoorthi — APPROVED** [review]

**Pull request merged**